evalml/data_checks/target_leakage_data_check.py

```diff
@@ -1,69 +1,70 @@
+    Correlation metrics available can be found in Woodwork's `dependence_dict method <https://woodwork.alteryx.com/en/stable/generated/woodwork.table_accessor.WoodworkTableAccessor.dependence_dict.html#woodwork.table_accessor.WoodworkTableAccessor.dependence_dict>`_.
 
     Args:
         pct_corr_threshold (float): The correlation threshold to be considered leakage. Defaults to 0.95.
-        method (string): The method to determine correlation. Use 'mutual' for mutual information, otherwise 'pearson' for Pearson correlation. Defaults to 'mutual'.
+        method (string): The method to determine correlation. Use 'max' for the maximum correlation, or for specific correlation metrics, use their name (ie 'mutual_info' for mutual information, 'pearson' for Pearson correlation, etc).
+            Defaults to 'mutual_info'.
     """
 
-    def __init__(self, pct_corr_threshold=0.95, method="mutual"):
+    def __init__(self, pct_corr_threshold=0.95, method="mutual_info"):
         if pct_corr_threshold < 0 or pct_corr_threshold > 1:
             raise ValueError(
                 "pct_corr_threshold must be a float between 0 and 1, inclusive.",
             )
-        if method not in ["mutual", "pearson"]:
-            raise ValueError(f"Method '{method}' not in ['mutual', 'pearson']")
+        methods = CONFIG_DEFAULTS["correlation_metrics"]
+        if method not in methods:
+            raise ValueError(f"Method '{method}' not in {methods}")
         self.pct_corr_threshold = pct_corr_threshold
         self.method = method
 
-    def _calculate_pearson(self, X, y):
+    def _calculate_dependence(self, X, y):
         highly_corr_cols = []
-        X_num = X.ww.select(include=numeric_and_boolean_ww)
-        if (
-            y.ww.logical_type.type_string not in numeric_and_boolean_ww
-            or len(X_num.columns) == 0
-        ):
-            return highly_corr_cols
-        highly_corr_cols = [
-            label
-            for label, col in X_num.iteritems()
-            if abs(y.corr(col)) >= self.pct_corr_threshold
-        ]
-        return highly_corr_cols
-
-    def _calculate_mutual_information(self, X, y):
-        highly_corr_cols = []
-        for col in X.columns:
-            cols_to_compare = X.ww[[col]]
-            cols_to_compare.ww[str(col) + "y"] = y
-            mutual_info = cols_to_compare.ww.mutual_information()
-            if (
-                len(mutual_info) > 0
-                and mutual_info["mutual_info"].iloc[0] > self.pct_corr_threshold
-            ):
-                highly_corr_cols.append(col)
+        X2 = X.ww.copy()
+        target_str = "target_y"
+        while target_str in list(X2.columns):
+            target_str += "_y"
+        X2.ww[target_str] = y
+        try:
+            dep_corr = X2.ww.dependence_dict(
+                measures=self.method,
+                target_col=target_str,
+            )
+        except KeyError:
```

Comment:
Could you point to the part in woodwork that raises this `KeyError` just for my edification?

Reply:
Right now, if we passed an invalid target_col for a metric (ie a boolean column for `Pearson` or `Spearman`), then we get a `keyerror` for now. This [WW PR](https://github.com/alteryx/woodwork/pull/1534) will address this and remove this keyerror